src/config.js

```diff
@@ -1,39 +1,109 @@
 import collector from './collector';
+import packageConfig from './packageConfig';
+
 const { getHostname, getCollectorPath } = collector;
 
 module.exports = function setConfig(configObject) {
   const defaults = {
+    clientId: '',
+    debug: false,
     host: getHostname(),
-    path: getCollectorPath(),
-    clientId: process.env.IOPIPE_TOKEN || process.env.IOPIPE_CLIENTID || '',
-    debug: process.env.IOPIPE_DEBUG || false,
-    networkTimeout: 5000,
-    timeoutWindow: Number(process.env.IOPIPE_TIMEOUT_WINDOW) || 150,
     installMethod: 'manual',
-    plugins: []
+    networkTimeout: 5000,
+    path: getCollectorPath(),
+    plugins: [],
+    timeoutWindow: 150
   };
 
   const config = Object.assign({}, defaults);
+
+  const packageConf = packageConfig.getConfig();
+
+  // Override default config with package config variables
+  if (packageConf) {
+    Object.keys(config).forEach(key => {
+      if (typeof packageConf[key] === 'undefined') return;
+
+      switch (key) {
+        case 'url':
+          config.host = getHostname(packageConf[key]);
+          config.path = getCollectorPath(packageConf[key]);
+          break;
+
+        case 'debug':
+          config[key] =
+            typeof packageConf[key] === 'boolean'
+              ? packageConf[key]
+              : config[key];
+          break;
+
+        case 'plugins':
+          // Loads {"plugins": ["plugin1", "plugin2"]}
+          config[key] =
+            packageConf[key].constructor === Array
+              ? packageConf[key]
+                  .map(src => packageConfig.requireFromString(src))
+                  .filter(plugin => typeof plugin !== 'undefined')
+              : config[key];
+          // Loads {"plugins": {"plugin1": ["arg1", "arg2"]}}
+          config[key] =
+            typeof packageConf[key] === 'object'
+              ? Object.keys(packageConf[key])
+                  .map(pluginKey => {
+                    if (packageConf[key][pluginKey].constructor === Array)
+                      return packageConfig.requireFromString(
+                        pluginKey,
+                        packageConf[key][pluginKey]
+                      );
+                    return packageConfig.requireFromString(pluginKey);
+                  })
+                  .filter(plugin => typeof plugin !== 'undefined')
+              : config[key];
+          break;
+
+        case 'timeoutWindow':
+          config[key] = Number.isInteger(packageConf[key])
+            ? packageConf[key]
+            : config[key];
+          break;
+
+        default:
+          config[key] = packageConf[key];
+      }
+    });
+  }
+
+  // Override default and package config with environment variables
+  config.clientId =
+    process.env.IOPIPE_TOKEN || process.env.IOPIPE_CLIENTID || config.clientId;
```

Comment:
I get the approach here, and maybe the original issue is that `defaults` is poorly named, or perhaps we should start with a blank object and add onto it rather than use "defaults." Suggestion not prescription but I do think we need to do something more concise